### ISSUE TITLE
Update Get-MessageIDs with Download Flag to Use Subject as File Name

### ISSUE THREAD

**Issue by TimberJank** [invictus-ir/Microsoft-Extractor-Suite#46]
Would it be possible to update Get-MessageIDs using yes with the  -Download argument so that the emails when downloaded have the subject name as the file name. Currently it appears the filename is Base64 but isnt decoding properly. Probably a user error. 

![image](https://github.com/invictus-ir/Microsoft-Extractor-Suite/assets/66707167/e7c64fe7-d4ec-4bd9-bcfa-65e4898f58cd)

**Comment by JoeyInvictus** [comment]
Hi @TimberJank, Thank you for the suggestion. Currently, the file name corresponds to the message ID. Initially, we attempted to use the subject name. However, numerous subject names were invalid file names, causing a good amount of errors. (
Example: "FW:" or "FWD:", which you often see after an email has been forwarded, is not a valid file name due to the colon ":")

This is certainly something we want to change to enhance user-friendliness, although we're uncertain about the best approach. While using the subject might be feasible, it would require lots of error handling. If you have any suggestions, please feel free to share them.

**Comment by JoeyInvictus** [comment]
I currently did the following to improve the output:

Replaced all non-valid filename characters with an underscore.
`$subject = $subject -replace '[\\/:*?"<>|]', '_'`

In addition, I added the received date and time before the filename. This ensures that all attachments are downloaded and not ignored if a filename already exists in the output directory. The output format is as follows:

`yyyyMMdd_HHmmss-Subject`

Let me know your thoughts on this approach. If it works as expected I can incorporate it into the get-emails function as well.

![image](https://github.com/invictus-ir/Microsoft-Extractor-Suite/assets/129975292/fef76147-ec3d-4c6e-8b13-a7dd90f36b40)
